components/post_view/post_list.jsx

```diff
@@ -173,24 +178,61 @@ export default class PostList extends React.PureComponent {
     }
 
     static getDerivedStateFromProps(props, state) {
-        const postListIds = props.postListIds || [];
-        let newPostListIds;
-
-        if (state.atEnd) {
-            newPostListIds = [...postListIds, PostListRowListIds.CHANNEL_INTRO_MESSAGE];
-        } else if (props.postVisibility >= Constants.MAX_POST_VISIBILITY) {
-            newPostListIds = [...postListIds, PostListRowListIds.MAX_MESSAGES_LOADED];
-        } else if (state.autoRetryEnable) {
-            newPostListIds = [...postListIds, PostListRowListIds.MORE_MESSAGES_LOADER];
-        } else {
-            newPostListIds = [...postListIds, PostListRowListIds.MANUAL_TRIGGER_LOAD_MESSAGES];
+        const {prevPropsPostListIds, postListIds: statePostListIds} = state;
+        const propsPostListIds = props.postListIds;
+        let postListIds = props.postListIds || [];
+
+        if (propsPostListIds) {
```

Comment:
I don't have any specific suggestions for how to improve this, but this block is really complicated and hard to understand.

Reply:
I agree